sc-iavm/api/SC-API.psm1

```diff
@@ -0,0 +1,1479 @@
+    }
+    if ($clearAssets) { $dict += @{ "assets" = @() } }
+
+    # We must have something to change before we send (more validation)
+    if ($dict.Count -eq 0) {
+        throw "A scan setting must be edited during a call to SC-Edit-Scan; no settings provided (`$dict is empty)"
+    }
+
+    #Write-Host ($dict | ConvertTo-Json -Depth 10 -Compress)
+    return SC-Connect -scResource scan -scResourceID $id -scHTTPMethod PATCH -scJSONInput $dict
+}
+
+
+function SC-Get-RolloverScans() {
+    <# Find any rollover scans #>
+    $resp = SC-Get-ScanInfo -filter managable -name -schedule -ownerGroup -owner -createdTime
+    return $resp.response.manageable | Where-Object { $_.schedule.type -eq "rollover" }
+}
+
+
+function SC-Purge-RolloverScans() {
+    <# Purge any and all rollover scans that are detected in the SecurityCenter #>
+    $rollover_scans = SC-Get-RolloverScans
+    $curr_count = 0
+    foreach ($scan in $rollover_scans) {
+        $progress = ($curr_count++ / $rollover_scans.Count) * 100
+        Write-Progress -Activity "Purging all Rollover scans" -Status ("Percent complete: " + $progress + "%") -PercentComplete $progress
+        Write-Host ("Purging: {id = " + $scan.id + "; name = `"" + $scan.name + "`"; owner = `"" + $scan.owner.username + "`"}")
+        SC-Delete-Scan -scan_id $scan.id | Out-Null
+    }
+}
+
+
+function SC-Upload-File() {
+    <#
+        A semi-loosely documented endpoint. It's documented, just not for all use-cases.
+        https://support.tenable.com/support-center/cerberus-support-center/includes/widgets/sc_api/File.html
+    #>
+    param(
+        [Parameter(Mandatory=$true)]
+        $filePath
+    )
+    # Write-Host $filePath
+    # Read in the entire file
+    $fileBin = [IO.File]::ReadAllBytes($filePath)
+    # Safely encode the file for transfer
+    $fileEnc = [System.Text.Encoding]::GetEncoding("ISO-8859-1").GetString($fileBin)
+    # Make a boundary to deliniate where the file information is
+    $boundary = [System.Guid]::NewGuid().ToString()
+    $LF = "`r`n"
+    $fileName = (Split-Path -Leaf $filePath)
+    # Manually build the request payload, doing something like ``@(foo,bar,baz) -join $LF`` adds spaces in spots and mucks it up.
+    $uploadBody = "----------$boundary" + $LF
+    $uploadBody += "Content-Disposition: form-data; name=`"Filedata`"; filename=`"$fileName`"$LF"
+    $uploadBody += "Content-Type: application/octet-stream$LF$LF"
+    $uploadBody += $fileEnc + $LF
+    $uploadBody += "----------$boundary--"
+    # Add in the additional headers required for this API endpoint
+    $additionalHeaders = @{"Content-Type"="multipart/form-data; boundary=--------$boundary"}
+    SC-Connect -scResource file/upload -scHTTPMethod POST -scAdditionalHeadersDict $additionalHeaders -scRawRequestPayload $uploadBody
+    # The name of the file on the SecurityCenter server to be used for other actions (such as importing)
+    return $script:scResponse.response.filename
```

Comment:
Fixed locally, but this method is the old variant of getting the response information. Store ``SC-Connect``, get the ``.response.filename``